data-integration/main.bal

```diff
@@ -0,0 +1,95 @@
+import data_integration.store;
+import ballerina/http;
+import ballerina/persist;
+
+final store:Client dbClient = check new ();
+
+service / on new http:Listener(9090) {
+    resource function post employees(store:EmployeeInsert employee)
+            returns http:Created|http:Conflict|http:InternalServerError {
+        int[]|persist:Error result = dbClient->/employees.post([employee]);
+        if result is persist:AlreadyExistsError {
+            return http:CONFLICT;
+        }
+        if result is persist:Error {
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return http:CREATED;
+    }
+
+    resource function post task(store:TaskInsert task)
+            returns http:Created|http:Conflict|http:InternalServerError {
+        int[]|persist:Error result = dbClient->/tasks.post([task]);
+        if result is persist:AlreadyExistsError {
+            return http:CONFLICT;
+        }
+        if result is persist:Error {
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return http:CREATED;
+    }
+
+    resource function get task/[int taskId]() returns http:Ok|http:NotFound|http:InternalServerError {
+        store:Task|persist:Error task = dbClient->/tasks/[taskId];
+        if task is persist:Error {
```

Comment:
Shall we switch the check and return early since we are not doing anything else on success? May apply to other resources too.